ios/Classes/AblyInstanceStore.m

```diff
@@ -0,0 +1,77 @@
+@import Ably;
+
+#import "AblyInstanceStore.h"
+#import <ably_flutter/ably_flutter-Swift.h>
+
+@implementation AblyInstanceStore {
+    NSMutableDictionary<NSNumber *, ARTRealtime *>* _realtimeInstances;
+    NSMutableDictionary<NSNumber *, ARTRest *>* _restInstances;
+    NSMutableDictionary<NSNumber *, ARTPaginatedResult *>* _paginatedResults;
+    long long _nextHandle;
+}
+
++ (instancetype)sharedInstance {
+    static AblyInstanceStore *sharedInstance = nil;
+    @synchronized(self) {
+        if (sharedInstance == nil) {
+            sharedInstance = [[self alloc] init];
+        }
+    }
+    return sharedInstance;
```

Comment:
I would recommend to use `dispatch_once()` more [here](https://stackoverflow.com/a/9119089)

Reply:
Thanks!: 5cac013e23e5dd54315de687c395b5fc9e2a1223